bin/cmsl1t_batch

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import click
+import click_log
+import collections
+import yaml
+import os
+from textwrap import dedent
+import logging
+from itertools import izip
+
+from cmsl1t.config import ConfigParser
+from cmsl1t.batch import Batch, condor_submit, lsf_submit
+from cmsl1t.batch import prepare_output_folders, get_config_name_template
+
+from cmsl1t.batch import create_run_script, create_info_file, prepare_jobs
+
+logger = logging.getLogger(__name__)
+click_log.basic_config(logger)
+
+
+@click.command()
+@click.option('-c', '--config_file', help='YAML style config file', type=click.File(), required=True)
+@click.option('-f', '--files-per-job', help='Give each job this many files', type=int, default=1)
+@click.option('--debug/--no-debug', help='Debug mode for the job submission', default=False)
+@click.option('--batch', default=Batch.condor, type=click.Choice([Batch.lsf, Batch.condor]),
+              help='Select the job submission system to use')
+def run(config_file, debug, batch, files_per_job):
+    if batch == Batch.lsf:
+        logger.warn('Legacy LSF system is no longer supported for cmsl1t.')
+        logger.warn(
+            ' see http://information-technology.web.cern.ch/services/batch')
+    # Read the config file
+    config = ConfigParser()
+    config.read(config_file)
+
+    # Get the output directory
+    output_folder = config.get('output', 'folder')
+    batch_dir, batch_config_dir, batch_log_dir = prepare_output_folders(
+        output_folder)
+
+    # Sort out a name for the batch config files
+    batch_filename = get_config_name_template(config_file, batch_config_dir)
+    # Prepare input jobs
+    outdir = os.path.join(batch_dir, "job_{index}")
+    job_generator = prepare_jobs(config, batch_filename, outdir, files_per_job)
+    job_configs, job_ids, output_folders = izip(*job_generator)
+
+    project_root = os.environ["PROJECT_ROOT"]
```

Comment:
Not specific to this PR, but it's bothered me for a while: is `PROJECT_ROOT` not quite vague?  I feel it should be something more like `CMSL1T_ROOT` or `CMSL1T_PROJECT_ROOT`?  Worth an issue for this?

Reply:
Yes, absolutely. It is something that has survived the prototype stage ;)